sources/packages/api/src/cra-integration/cra-integration.service.ts

```diff
@@ -0,0 +1,145 @@
+import { Injectable, Scope } from "@nestjs/common";
+import { ConfigService } from "../services";
+import { SshService } from "../services/ssh/ssh.service";
+import * as Client from "ssh2-sftp-client";
+import {
+  CRAPersonRecord,
+  CRAUploadResult,
+  TransactionCodes,
+} from "./cra-integration.models";
+import { CRAIntegrationConfig, SFTPConfig } from "../types";
+import { CRAFileHeader } from "./cra-files/cra-file-header";
+import { CRAFileFooter } from "./cra-files/cra-file-footer";
+import { CRAFileLine } from "./cra-files/cra-file";
+import { CRAFileIVRequestRecord } from "./cra-files/cra-file-request-record";
+
+/**
+ * Manages the creation of the content files that needs to be sent
+ * to Canada Revenue Agency (CRA). These files are created based
+ * on a fixed size format and uploaded to a SFTP on Government
+ * ZONE B network for further processing and final send to CRA servers.
+ */
+@Injectable()
+export class CRAIntegrationService {
+  private readonly craConfig: CRAIntegrationConfig;
+  private readonly ftpConfig: SFTPConfig;
+
+  constructor(config: ConfigService, private readonly sshService: SshService) {
+    this.craConfig = config.getConfig().CRAIntegration;
+    this.ftpConfig = config.getConfig().zoneBsFTP;
+  }
+
+  /**
+   * Creates a matching run request used, for instance, to validate
+   * the SIN information.
+   * @param records Personal/individual records to be processed.
+   * @param sequence File request sequence number that is required
+   * by CRA server processing.
+   * @returns Matching run content.
+   */
+  public createMatchingRunContent(
+    records: CRAPersonRecord[],
+    sequence: number,
+  ): CRAFileLine[] {
+    const processDate = new Date();
+    const craFileLines: CRAFileLine[] = [];
+
+    // Header
+    const fileHeader = this.createHeader(
+      TransactionCodes.MatchingRunHeader,
+      processDate,
+      sequence,
+    );
+    craFileLines.push(fileHeader);
+    // Records
+    const fileRecords = records.map((r) => {
```

Comment:
if possible can you change `r` to something meaningful

Reply:
Changed!